test/oauth.spec.ts

```diff
@@ -0,0 +1,64 @@
+import "mocha";
+import {ApiCall} from "../src/routes/oath";
+//import {refresh_token} from "../src/routes/oath";
+const chai = require( 'chai')
+var assert = chai.assert;
+const axios = require("axios");
+
+const HAPIKEY = process.env.HAPIKEY;
+
+const correctCall = async () => {
+    try {
+      //  const allContacts = 'https://api.hubapi.com/contacts/v1/lists/all/contacts/all?hapikey=' + HAPIKEY;
+        const response = await axios.get('https://dog.ceo/api/breeds/list/all')
+        //const response = await axios.get(allContacts);
+        const data = response.data;
+        return data;
+    } catch (e) {
+        console.error('  > Unable to retrieve contact');
+        return JSON.parse(e.response.body);
+    }
+
+}
+
+const errorCall = async () => {
+    try {
+          const allContacts = 'https://api.hubapi.com/contacts/v1/lists/all/contacts/all?hapikey=' + HAPIKEY;
+          const response = await axios.get(allContacts);
+          const data = response.data;
+          return data;
+    } catch (e) {
+        console.error('  > Unable to retrieve contact');
+        return JSON.parse(e.response.body);
+    }
+
+}
+
+it(' If the API call returns no errrors then getHubspotAccessfunc is not called ',async function()
+{
+    const res = await ApiCall(correctCall,"wrongtoken");
+    assert(res.status == 'success');
+
+})
+
+
+it(' If the api call returns an error and an invalid refresh token is passed then "BAD_REFRESH_TOKEN" error is returned  ',async function()
```

Comment:
I ran tests on my machine and this test fails:

 2 passing (1s)
  1 failing

  1)  If the api call returns an error and an invalid refresh token is passed then "BAD_REFRESH_TOKEN" error is returned  :
     TypeError: Cannot read property 'status' of undefined
      at Context.<anonymous> (test/oauth.spec.ts:49:25)
      at step (test/oauth.spec.ts:33:23)
      at Object.next (test/oauth.spec.ts:14:53)
      at fulfilled (test/oauth.spec.ts:5:58)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

Reply:
Could you please check if you have an environment variable NODE_ENV= 'test'?

Reply:
I get the same error with NODE_ENV= 'test'.

Reply:
Could you please try console logging NODE_ENV in your code? Perhaps it wasn't picked up (only reason I'm saying is because I had an issue where I had all the env variables but some were not being read before). The test is passing for me and the only reason I see for this error is for the if(process.node_env == test) to be missed.

Reply:
not sure, i logged it and it says test

Reply:
Perhaps we can debug this during lab.